source/elements/oneMKL/source/architecture/other_architecture.inc.rst

```diff
@@ -11,10 +11,41 @@ This section covers all other features in the design of oneMKL architecture.
 
 .. _onemkl_spec_current_version:
 
-Current Version of this oneMKL Specification
+Specification Version and Compliance
 +++++++++++++++++++++++++++++++++++++++++++++
 
-This is the oneMKL specification which is part of the oneAPI specification version 1.0.0.
+Each oneMKL domain must define a preprocessor macro to represent the version of the specification that the implementation is compliant with.
+
+The macros for each domain are listed as follows:
+
+  | ONEMKL_BLAS_SPEC_VERSION
+  | ONEMKL_LAPACK_SPEC_VERSION
+  | ONEMKL_SPBLAS_SPEC_VERSION
+  | ONEMKL_DFT_SPEC_VERSION
+  | ONEMKL_RNG_SPEC_VERSION
+  | ONEMKL_STAT_SPEC_VERSION
+  | ONEMKL_VM_SPEC_VERSION
+
+The specification version can be created by appending all digits of the specification version in the format of <MAJOR><MINOR>. MINOR version always uses two digits. This version can be used to check the compatibility of the implementation with the specification version. Note that the revision is not included here because it reflects changes only for the specification document without affecting the implementation. If the implementation is not compliant with any release of the specification, then the macro must have a numerical value of `000`.
+
+Version Example
+
+  | oneAPI 1.1 rev 1 will be represented as a numerical value of 101
+  | oneAPI 1.2 rev 1 will be represented as a numerical value of 102
+  | oneAPI 1.2 rev 2 will be represented as a numerical value of 102
+
+Macro Example
+
+.. code-block:: c
+
+  // For oneAPI 1.2 rev 1
```

Comment:
Question: Can domain be aligned with provisional spec version? 
E.g. the latest available release is 1.4. provisional https://github.com/uxlfoundation/oneAPI-spec/releases 

If yes, how we should update the macro in this case?

Reply:
@paveldyakov the provisional versioning (https://github.com/uxlfoundation/oneAPI-spec/blob/main/doc/versioning.rst#provisional-versions) is for specifications that are pending approval.

It is highly unlikely that we will have implementations that will comply with provisional specifications unless the approval of the spec happens later than the implementation in the backend library. If the backend is ready before the spec, the macro can point to a future non-provisional revision because the spec release follows a predictable convention.

From the above link:
> A specification may be published before approval, but must be labeled provisional. Provisional specifications are published as a series of revisions until approval. After approval, the provisional label is removed and the rev is set to 1.

Reply:
sounds good, thanks!